### PR TITLE
Make GetSomeTestImage more varied

### PR DESCRIPTION
**Pull request by lvandeve**
Make the randomly generated image from GetSomeTestImage more varied:
before, it always had a quarter circle of the same size in the top left
corner, and random noise elsewhere.

Now, it has a circle with random radius at random location, a rectangle
with random noise, and a random colored gradient as background.

I needed this to generate a jxl image with blending that has the
blending visible (multiple circles at random locations overlaid, which
was not possible to see if circle was always the same), but
this is worth committing because it improves the tests in general.

### CODE REVIEW

File: lib/jxl/test_utils.h [hunk 1]
```diff
@@ -273,25 +273,60 @@ std::vector<ColorEncodingDescriptor> AllEncodings() {
 std::vector<uint8_t> GetSomeTestImage(size_t xsize, size_t ysize,
                                       size_t num_channels, uint16_t seed) {
   // Cause more significant image difference for successive seeds.
-  std::mt19937 rng(seed);
-  std::uniform_int_distribution<uint16_t> dark(0, 32767);
+  std::mt19937 std_rng(seed);
+  std::uniform_int_distribution<uint16_t> std_distr(0, 65535);
+
+  // Returns random integer in interval (0, max_value - 1)
```
Comment (jan-wassenberg) [inline lib/jxl/test_utils.h:279]: Nit: uniform_int_distribution takes a closed interval, so this is in [0, max_value).

File: lib/jxl/test_utils.h [hunk 2]
```diff
@@ -273,25 +273,60 @@ std::vector<ColorEncodingDescriptor> AllEncodings() {
...
       // put some shape in there for visual debugging
-      if (x * x + y * y < 1000) {
+      if ((x - circle_x) * (x - circle_x) + (y - circle_y) * (y - circle_y) <
+          circle_r * circle_r) {
         r = (65535 - x * y) ^ seed;
         g = (x << 8) + y + seed;
         b = (y << 8) + x * seed;
         a = 32768 + x * 256 - y;
+      } else if (x > rect_x0 && x < rect_x1 && y > rect_y0 && y < rect_y1) {
+        r = rng(65536);
```
Comment (jan-wassenberg) [inline lib/jxl/test_utils.h:326]: This could actually overflow u16 if the RNG returns the max value 65535.

File: lib/jxl/test_utils.h (hunk 1 above)
Comment (lvandeve) [reply lib/jxl/test_utils.h:279]: Thanks for spotting that, dividing through 65536.0f instead should have fixed this

File: lib/jxl/test_utils.h (hunk 2 above)
Comment (lvandeve) [reply lib/jxl/test_utils.h:326]: Thanks for spotting that! The fix from the other comment should have fixed this

File: lib/jxl/test_utils.h (hunk 1 above)
Comment (jan-wassenberg) [reply lib/jxl/test_utils.h:279]: Sounds good, thanks for making the change!

**Review by jan-wassenberg — APPROVED** [review]

**Pull request merged**